### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by hanhanW**
* Example Usage
* Installation
* Supported Features

---
This change is [Reviewable](https://reviewable.io/reviews/peter50216/pwntools-ruby/47)

### CODE REVIEW

**Comment by hanhanW** [comment]
Where should we publish the documentation?
@peter50216 @david942j

**Comment by david942j** [comment]
http://www.rubydoc.info/github/peter50216/pwntools-ruby/master/frames ?

File: README.md [hunk 1]
```diff
@@ -32,6 +29,54 @@ p pack(0x41424344)  # 'ABCD'
 context.local(bits: 16) do
   p pack(0x4142)  # 'AB'
 end
+
+context.log_level = :debug
+s = Sock.new('exploitme.example.com', 31337)
```
Comment (david942j) [inline README.md]: Use example for real challenge
The easiest problem on pwnable.kr or pwnable.tw

File: README.md [hunk 2]
```diff
@@ -25,13 +23,63 @@ Would try to have consistent naming with original pwntools, and do things in Rub
...
+gem build keystone.gemspec
+gem install keystone
+```
+
+Some of the features (assembling/disassembling) require non-Ruby dependencies. Checkout the
+installation guide for
+[keystone-engine](https://github.com/keystone-engine/keystone/tree/master/docs) and
+[capstone-engine](http://www.capstone-engine.org/documentation.html).
+
+Or you can be able to get running quickly with
```
Comment (peter50216) [inline README.md]: "can" and "be able to" are duplicated.

**Comment by hanhanW** [comment]
What happened to the url of `documentation` here ...?

**Comment by david942j** [comment]
Suggestion:
Add check list of supported features in readme
Ex.
Architectures:
* [X] i386
* [X] amd64
* [ ] arm

Modules:
* [X] asm
* [X] disasm
* [X] shellcraft

File: README.md [hunk 3]
```diff
@@ -12,26 +12,77 @@ Always sad when playing CTF that there's nothing equivalent to pwntools in Pytho
...
 # encoding: ASCII-8BIT
 # The encoding line is important most time, or you'll get "\u0000" when using "\x00" in code,
 # which is NOT what we want when doing pwn...
 
 require 'pwn'
 
-p pack(0x41424344)  # 'DCBA'
-context.endian = 'big'
-p pack(0x41424344)  # 'ABCD'
+context.log_level = :debug
```
Comment (david942j) [inline README.md]: set context.arch = 'i386'

File: README.md [hunk 4]
```diff
@@ -12,28 +12,116 @@ Always sad when playing CTF that there's nothing equivalent to pwntools in Pytho
...
 So this is an attempt to create such library.
 
-There's almost NOTHING here now.
-(Edit: there's something here now, but not much :wink:)
-Going to implement important things (socket, tubes, asm/disasm, pack/unpack utilities) first.
 Would try to have consistent naming with original pwntools, and do things in Ruby style.
 
 # Example Usage
+
+Here's an exploitation for `start` which is a challenge on `pwnable.tw`.
```
Comment (david942j) [inline README.md]: change `pwnable.tw` to [pwnable.tw](https://pwnable.tw])

File: README.md [hunk 5]
```diff
@@ -12,28 +12,116 @@ Always sad when playing CTF that there's nothing equivalent to pwntools in Pytho
...
+cd build
+../make-share.sh
+sudo make install
+```
+
+# Supported Features
+
+## Architectures
+
+- [x] i386
```
Comment (david942j) [inline README.md]: I think list arm and thumb in unchecked items is enough

File: README.md [hunk 6]
```diff
@@ -12,28 +12,116 @@ Always sad when playing CTF that there's nothing equivalent to pwntools in Pytho
...
+- [ ] ia64
+- [ ] mips
+- [ ] powerpc
+- [ ] sparc
+- [ ] thumb
+- [ ] vax
+
+## Modules
+
+- [ ] adb
```
Comment (david942j) [inline README.md]: - [x] context
- [x] asm
- [x] disasm
- [x] shellcraft
- [x] elf
- [x] dynelf
- [x] logger
- [ ] tube
  - [x] sock
  - [ ] process
- [ ] fmtstr
- [x] util
  - [x] pack
  - [x] cyclic
  - [x] fiddling

**Comment by hanhanW** [comment]
Review status: 0 of 1 files reviewed at latest revision, 6 unresolved discussions.

---

*[README.md, line 34 at r1](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/47#-Ku7hljpIdf29Zn3mqNJ-r1-34:-KuJsrSghkWEgEs4H4N3:b-896fix) ([raw file](https://github.com/peter50216/pwntools-ruby/blob/5955714aafe3a0894b314e04f0d72b497a70e61d/README.md#L34)):*
<details><summary><i>Previously, david942j wrote…</i></summary><blockquote>

Change example to a real challenge
The easiest problem on pwnable.kr or pwnable.tw
</blockquote></details>

Done.

---

*[README.md, line 70 at r2](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/47#-Ku7hljpIdf29Zn3mqNJ-r2-70:-KuJssHjAjr1-r0Rf-G8:b-896fix) ([raw file](https://github.com/peter50216/pwntools-ruby/blob/5d35890667a94e74d9693b584b5c619710b612eb/README.md#L70)):*
<details><summary><i>Previously, peter50216 (Peter Shih) wrote…</i></summary><blockquote>

"can" and "be able to" are duplicated.
</blockquote></details>

Done.

---

*[README.md, line 28 at r5](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/47#-Ku7hljpIdf29Zn3mqNJ-r5-28:-KuJsqJbiBxrxPV6cUDl:b-896fix) ([raw file](https://github.com/peter50216/pwntools-ruby/blob/751c26a1aec5501caa0bb428757f06257da60b8a/README.md#L28)):*
<details><summary><i>Previously, david942j wrote…</i></summary><blockquote>

set context.arch = 'i386'
</blockquote></details>

Done.

---

*[README.md, line 19 at r6](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/47#-Ku7hljpIdf29Zn3mqNJ-r6-19:-KuJsp62NUOaRY1mALY-:b-896fix) ([raw file](https://github.com/peter50216/pwntools-ruby/blob/0753b8bba04c36fdc5efa72c19ef83e4f794d88f/README.md#L19)):*
<details><summary><i>Previously, david942j wrote…</i></summary><blockquote>

change `pwnable.tw` to [pwnable.tw](https://pwnable.tw)
</blockquote></details>

Done.

---

*[README.md, line 93 at r6](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/47#-Ku7hljpIdf29Zn3mqNJ-r6-93:-KuJst15GLYT-W3mLxd-:b-896fix) ([raw file](https://github.com/peter50216/pwntools-ruby/blob/0753b8bba04c36fdc5efa72c19ef83e4f794d88f/README.md#L93)):*
<details><summary><i>Previously, david942j wrote…</i></summary><blockquote>

I think list arm and thumb in unchecked items is enough
</blockquote></details>

Done.

---

*[README.md, line 108 at r6](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/47#-Ku7hljpIdf29Zn3mqNJ-r6-108:-KuJstcrVGJrDwFPoqs5:b-896fix) ([raw file](https://github.com/peter50216/pwntools-ruby/blob/0753b8bba04c36fdc5efa72c19ef83e4f794d88f/README.md#L108)):*
<details><summary><i>Previously, david942j wrote…</i></summary><blockquote>

- [x] context
- [x] asm
- [x] disasm
- [x] shellcraft
- [x] elf
- [x] dynelf
- [x] logger
- [ ] tube
  - [x] sock
  - [ ] process
- [ ] fmtstr
- [x] util
  - [x] pack
  - [x] cyclic
  - [x] fiddling
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/47)*

**Comment by david942j** [comment]
LGTM

**Comment by david942j** [comment]
Reviewed 1 of 1 files at r7.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/47)*

**Comment by david942j** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/47#-:-KuP9YyHDfkNoRcvqKBj:bnfp4nl)*

**Comment by david942j** [comment]
LGTM

**Pull request merged**